pandas/io/parsers.py

```diff
@@ -2876,7 +2876,7 @@ def _check_for_bom(self, first_row):
             return [new_row] + first_row[1:]
 
         elif len(first_row_bom) > 1:
-            return [first_row_bom[1:]]
+            return [first_row_bom[1:]] + first_row[1:]
```

Comment:
Hmm I don't find this very clear why we would do this - can you try refactoring the code above this to better suit the requirement?

Reply:
from what I understand
the code block above removes the BOM from the first element of the row and removes the quotes if it was a quoted string. i.e. if it was of the format `<BOM><QUOTE>abc<QUOTE>def` then it extracts out `abcdef`

I've tried to refactor it slightly